tools/autograd/gen_autograd.py

```diff
@@ -64,11 +64,16 @@
     'sparse_coo_tensor_with_dims_and_tensors': 'values',
 }
 
+VIEW_FUNCTIONS_WITH_METADATA_CHANGE = ['view_as_real', 'view_as_complex']
+
+for key in VIEW_FUNCTIONS_WITH_METADATA_CHANGE:
+    VIEW_FUNCTIONS[key] = 'self'
```

Comment:
nit for consistency, I would have done the same as what is done below for `RETURNS_VIEWS_OF_INPUT` and just add the two.

Reply:
but `real` and `imag` are composite view functions and they are not different from chunk or split in any way so I don't see why we need to do that

Reply:
I didn't mean that you should change these lists at all.
It is just that we have 3 lists that contain the previous one. So it would be simpler to just create them in that order.
But we can re-order that later if you prefer.